tests/appsec/test_automated_login_events.py

```diff
@@ -1348,3 +1218,703 @@ def validate_iden(meta):
+                assert meta["usr.id"] == "sdkUser"
+                assert meta["_dd.appsec.usr.id"] == "social-security-id"
+
+    def setup_login_sdk_failure_local(self):
+        self.r_sdk_failure = weblog.post(
+            "/login?auth=local&sdk_event=failure&sdk_user=sdkUser&sdk_user_exists=true",
+            data=login_data(context, INVALID_USER, PASSWORD),
+        )
+
+    def test_login_sdk_failure_local(self):
+        assert self.r_sdk_failure.status_code == 401
+        for _, trace, span in interfaces.library.get_spans(request=self.r_sdk_failure):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.failure.usr.login"] == "sdkUser"
+            assert meta["_dd.appsec.usr.login"] == INVALID_USER
+            assert meta["_dd.appsec.events.users.login.failure.auto.mode"] == "identification"
+            assert meta["appsec.events.users.login.failure.track"] == "true"
+            assert meta["_dd.appsec.events.users.login.failure.sdk"] == "true"
+            assert meta["appsec.events.users.login.failure.usr.exists"] == "true"
+
+    def setup_login_sdk_failure_basic(self):
+        self.r_sdk_failure = weblog.get(
+            "/login?auth=basic&sdk_event=failure&sdk_user=sdkUser&sdk_user_exists=true",
+            headers={"Authorization": BASIC_AUTH_INVALID_USER_HEADER},
+        )
+
+    @missing_feature(context.library == "php", reason="Basic auth not implemented")
+    def test_login_sdk_failure_basic(self):
+        assert self.r_sdk_failure.status_code == 401
+        for _, trace, span in interfaces.library.get_spans(request=self.r_sdk_failure):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.failure.usr.login"] == "sdkUser"
+            assert meta["_dd.appsec.usr.login"] == INVALID_USER
+            assert meta["_dd.appsec.events.users.login.failure.auto.mode"] == "identification"
+            assert meta["appsec.events.users.login.failure.track"] == "true"
+            assert meta["_dd.appsec.events.users.login.failure.sdk"] == "true"
+            assert meta["appsec.events.users.login.failure.usr.exists"] == "true"
+
+    def setup_signup_local(self):
+        self.r_success = weblog.post("/signup", data=login_data(context, NEW_USER, PASSWORD))
+
+    def test_signup_local(self):
+        assert self.r_success.status_code == 200
+        for _, trace, span in interfaces.library.get_spans(request=self.r_success):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.signup.usr.login"] == NEW_USER
+            assert meta["_dd.appsec.usr.login"] == NEW_USER
+            assert meta["_dd.appsec.events.users.signup.auto.mode"] == "anonymization"
```

Comment:
isn't this supposed to be non anonymized class ?
```suggestion
            assert meta["_dd.appsec.events.users.signup.auto.mode"] == "identification"
```